tlair/src/main/scala/at/forsyte/apalache/tla/types/TypeUnifier.scala

```diff
@@ -223,6 +223,11 @@ class TypeUnifier(varPool: TypeVarPool) {
       case (RecRowT1(RowT1(lfields, lv)), RecRowT1(RowT1(rfields, rv))) =>
         unifyRows(lfields, rfields, lv, rv).map(t => RecRowT1(t))
 
+      case (rec @ RecT1(_), rowRec @ RecRowT1(_)) => compute(rowRec, rec)
+
+      // An old record type can be treated as a monomorphic row-typed record
+      case (rowRec @ RecRowT1(_), RecT1(fields)) => compute(rowRec, RecRowT1(RowT1(fields, None)))
+
```

Comment:
The motivation here was that, at some point for a reason I'm not totally clear on, operations in https://github.com/informalsystems/apalache/tree/cbf3870476b035a3280a9cb790edd60803b17a6f/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SymbStateDecoder.scala are decoding an Arena that must have been a row-typed record at type checking into an old record, and then passing that on to the builder, which then fails to unify the types.

However, I think this addition is well motivated in any case, since an old record typing should be able to be considered a subtype in this way. This basically picks up a piece of the logic from the `commontSuperType` code that is being removed.